community/lucene-index/src/main/java/org/neo4j/index/impl/lucene/legacy/LuceneDataSource.java

```diff
@@ -111,6 +115,45 @@ public String toString()
     private boolean closed;
     private LuceneFilesystemFacade filesystemFacade;
     private IndexReferenceFactory indexReferenceFactory;
+    private Map<IndexIdentifier,Map<String,DocValuesType>> indexTypeMap;
+
+    public void assertValidType( String key, Object value, IndexIdentifier identifier )
+    {
+        DocValuesType expectedType;
+        String expectedTypeName;
+        if ( value instanceof Number )
+        {
+            expectedType = DocValuesType.SORTED_NUMERIC;
+            expectedTypeName = "numbers";
+        }
+        else
+        {
+            expectedType = DocValuesType.SORTED_SET;
+            expectedTypeName = "strings";
+        }
+        Map<String,DocValuesType> stringDocValuesTypeMap = indexTypeMap.get( identifier );
```

Comment:
indexTypeMap.computeIfAbsent?

Reply:
Sadly not really applicable here, since we need to load the index searcher, and that will populate the map for us.